prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c

```diff
@@ -75,35 +75,97 @@ fi_ibv_rdm_start_connection(struct fi_ibv_rdm_ep *ep,
 	return FI_SUCCESS;
 }
 
-ssize_t fi_ibv_rdm_start_disconnection(struct fi_ibv_rdm_conn *conn)
+static inline ssize_t
+fi_ibv_rdm_disconnect_request(struct fi_ibv_rdm_ep *ep,
+			      struct fi_ibv_rdm_conn *conn)
+{
+	struct fi_ibv_rdm_buf *sbuf = NULL;
+	ssize_t ret = FI_SUCCESS;
+
+	if (!ep->is_closing) {
+		pthread_mutex_lock(&ep->cm_lock);
+	}
```

Comment:
This sort of locking looks racy to me.  There's nothing protecting the is_closing variable, which makes me think it could change between the check here and the check below where the lock is released.